pkg/processor/build/builder.go

```diff
@@ -324,9 +331,30 @@ func (b *Builder) resolveFunctionPath(functionPath string) (string, error) {
 		return "", fmt.Errorf("Function path doesn't exist: %s", resolvedPath)
 	}
 
+	if util.IsCompressed(resolvedPath) {
+		resolvedPath, err = b.decompressFunctionArchive(resolvedPath)
+		if err != nil {
+			return "", errors.Wrap(err, "Failed to decompress function archive")
+		}
+	}
+
 	return resolvedPath, nil
 }
 
+func (b *Builder) decompressFunctionArchive(functionPath string) (string, error) {
+	// create a staging directory
+	tempDir, err := ioutil.TempDir("", "nuclio-decompress-")
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to create temp directory for decompressing archive %v", functionPath)
+	}
+
+	err = util.NewDecompressor(b.logger).Decompress(functionPath, tempDir)
```

Comment:
NewDecompressor must return an error as well. This is because sometimes as code evolves constructors may do something that returns an error and it's a bitch to refactor all the calls. Might as well make constructors always return error

Reply:
Fixed